client/static/js/dashboard_view.js

```diff
@@ -134,18 +134,25 @@ var DashboardView = function(userProfile) {
     var html = "";
     var x, y, severity;
     var serverId, groupId, numberOfTriggers;
+    var hostStatuses, hostStatus;
 
     for (x = 0; x < replyData["serverStatus"].length; ++x) {
       serverId = replyData["serverStatus"][x]["serverId"];
+      hostStatuses = replyData["serverStatus"][x]["hostStatus"];
       html += "<tr>";
       y = 0;
       for (groupId in replyData["serverStatus"][x]["hostgroups"]) {
+        hostStatus = hostStatuses[y];
         html += "<tr>"; // ==============  start of a row ================
         if (y == 0) {
           html += "<td rowspan='" + escapeHTML(parsedData[serverId]["numberOfHostgroups"]) + "'>";
           html += escapeHTML(replyData["serverStatus"][x]["serverNickname"]);
           html += "</td>";
         }
+        if (hostStatus["numberOfGoodHosts"] + hostStatus["numberOfBadHosts"] == 0) {
+          ++y;
+          continue;
```

Comment:
It seems that it breaks the "tr" element.
Although the <tr> tag is opened at the top of the block, it will never be closed when the "continue" is evaluated.
(It's closed at the end of the block).